### ISSUE TITLE
Failed test `Driver/darwin-ld-lto.c` in 13.0.1-rc2

### ISSUE THREAD

**Issue by tru** [llvm/llvm-project#53200]
While testing 13.0.1-rc2 I ran into the the following test result error:

```
FAIL: Clang :: Driver/darwin-ld-lto.c (7968 of 91780)
******************** TEST 'Clang :: Driver/darwin-ld-lto.c' FAILED ********************
Script:
--
: 'RUN: at line 5';   mkdir -p /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/tools/clang/test/Driver/Output/darwin-ld-lto.c.tmp/bin
: 'RUN: at line 6';   mkdir -p /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/tools/clang/test/Driver/Output/darwin-ld-lto.c.tmp/lib
: 'RUN: at line 7';   touch /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/tools/clang/test/Driver/Output/darwin-ld-lto.c.tmp/lib/libLTO.dylib
: 'RUN: at line 8';   /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/bin/clang -fuse-ld= -target x86_64-apple-darwin10 -### /Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c    -ccc-install-dir /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/tools/clang/test/Driver/Output/darwin-ld-lto.c.tmp/bin -mlinker-version=133 2> /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/tools/clang/test/Driver/Output/darwin-ld-lto.c.tmp.log
: 'RUN: at line 10';   /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/bin/FileCheck -check-prefix=LINK_LTOLIB_PATH /Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c -input-file /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/tools/clang/test/Driver/Output/darwin-ld-lto.c.tmp.log
: 'RUN: at line 17';   /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/bin/clang -fuse-ld= -target x86_64-apple-darwin10 -### /Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c    -ccc-install-dir /Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/dummytestdir -mlinker-version=133 2> /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/tools/clang/test/Driver/Output/darwin-ld-lto.c.tmp.log
: 'RUN: at line 19';   /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/bin/FileCheck -check-prefix=LINK_LTOLIB_PATH /Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c -input-file /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/tools/clang/test/Driver/Output/darwin-ld-lto.c.tmp.log
: 'RUN: at line 23';   /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/bin/clang -target x86_64-apple-darwin10 /Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c -flto=full -### 2>&1 |    /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/bin/FileCheck -check-prefix=FULL_LTO_OBJECT_PATH /Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c
: 'RUN: at line 28';   /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/bin/clang -target x86_64-apple-darwin10 /Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c -flto=thin -### 2>&1 |    /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/bin/FileCheck -check-prefix=THIN_LTO_OBJECT_PATH /Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c
: 'RUN: at line 36';   /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/bin/clang -target arm64-apple-darwin /Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c -flto -fglobal-isel -### 2>&1 |    /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/bin/FileCheck --check-prefix=GISEL /Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c
--
Exit Code: 1

Command Output (stderr):
--
/Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c:26:31: error: FULL_LTO_OBJECT_PATH-SAME: expected string not found in input
// FULL_LTO_OBJECT_PATH-SAME: "-object_path_lto"
                              ^
<stdin>:6:15: note: scanning from here
 "/usr/bin/ld" "-dynamic" "-arch" "x86_64" "-macosx_version_min" "10.6.0" "-o" "a.out" "-lcrt1.10.6.o" "/var/folders/1t/vx8d_cyn4517qwkq2s_lgvp80000gn/T/darwin-ld-lto-9474c9.o" "-lSystem" "/Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/lib/clang/13.0.1/lib/darwin/libclang_rt.osx.a"
              ^
<stdin>:6:16: note: possible intended match here
 "/usr/bin/ld" "-dynamic" "-arch" "x86_64" "-macosx_version_min" "10.6.0" "-o" "a.out" "-lcrt1.10.6.o" "/var/folders/1t/vx8d_cyn4517qwkq2s_lgvp80000gn/T/darwin-ld-lto-9474c9.o" "-lSystem" "/Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/lib/clang/13.0.1/lib/darwin/libclang_rt.osx.a"
               ^

Input file: <stdin>
Check file: /Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c

-dump-input=help explains the following input dump.

Input was:
<<<<<<
           1: clang version 13.0.1
           2: Target: x86_64-apple-darwin10
           3: Thread model: posix
           4: InstalledDir: /Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/bin
           5:  "/Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/bin/clang-13" "-cc1" "-triple" "x86_64-apple-macosx10.6.0" "-Wundef-prefix=TARGET_OS_" "-Werror=undef-prefix" "-Wdeprecated-objc-isa-usage" "-Werror=deprecated-objc-isa-usage" "-emit-llvm-bc" "-flto=full" "-flto-unit" "-disable-free" "-disable-llvm-verifier" "-discard-value-names" "-main-file-name" "darwin-ld-lto.c" "-mrelocation-model" "pic" "-pic-level" "2" "-mframe-pointer=all" "-fno-rounding-math" "-munwind-tables" "-faligned-alloc-unavailable" "-fcompatibility-qualified-id-block-type-checking" "-fvisibility-inlines-hidden-static-local-var" "-target-cpu" "core2" "-tune-cpu" "generic" "-mllvm" "-treat-scalable-fixed-error-as-warning" "-debugger-tuning=lldb" "-target-linker-version" "13.0.1" "-fcoverage-compilation-dir=/Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/tools/clang/test/Driver" "-resource-dir" "/Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/lib/clang/13.0.1" "-internal-isystem" "/usr/local/include" "-internal-isystem" "/Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/lib/clang/13.0.1/include" "-internal-externc-isystem" "/usr/include" "-fdebug-compilation-dir=/Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/tools/clang/test/Driver" "-ferror-limit" "19" "-stack-protector" "1" "-fblocks" "-fencode-extended-block-signature" "-fregister-global-dtors-with-atexit" "-fgnuc-version=4.2.1" "-fmax-type-align=16" "-D__GCC_HAVE_DWARF2_CFI_ASM=1" "-o" "/var/folders/1t/vx8d_cyn4517qwkq2s_lgvp80000gn/T/darwin-ld-lto-9474c9.o" "-x" "c" "/Users/tobias/code/llvm-releases/13.0.1/rc2/llvm-project/clang/test/Driver/darwin-ld-lto.c"
           6:  "/usr/bin/ld" "-dynamic" "-arch" "x86_64" "-macosx_version_min" "10.6.0" "-o" "a.out" "-lcrt1.10.6.o" "/var/folders/1t/vx8d_cyn4517qwkq2s_lgvp80000gn/T/darwin-ld-lto-9474c9.o" "-lSystem" "/Users/tobias/code/llvm-releases/13.0.1/rc2/Phase3/Release/llvmCore-13.0.1-rc2.obj/lib/clang/13.0.1/lib/darwin/libclang_rt.osx.a"
same:26'0                   X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
same:26'1                    ?

      possible intended match
>>>>>>

--
```

Can someone familiar with this test / code advise if this is a blocking issue?

**Comment by tstellar** [comment]
@tru Did this fail in rc1 too?

**Comment by tru** [comment]
@tstellar I went back in all the older testing logs I had and it turns out that I must have missed this earlier since this has failed all the way back in 13.0.0-rc4 (which is the oldest log I had saved).

So I would probably not call it a blocker. But it would be interesting to see if someone can figure out why this fails.

**Comment by tru** [comment]
This is fixed in 15.0.0-rc1.
